### PR TITLE
Add subset() method to Arr

### PR DESCRIPTION
**Pull request by Altreus**
We found ourselves needing a version of filter_keys that created them - useful for forms.

Will create docs if there are no objections to the method.

We considered writing it in terms of Arr::merge but we decided to keep it simple for the 90% case of handling input data.

### CODE REVIEW

File: classes/arr.php [hunk 1]
```diff
@@ -1207,4 +1207,24 @@ public static function next_by_value($array, $value, $get_value = true, $strict
...
+	 * @param   array    $keys     list of keys (or indices) to return
+	 * @param   mixed    $default  value of missing keys; default null
+	 *
+	 * @return  array  An array containing the same set of keys provided.
+	 */
+	public function subset($array, $keys, $default=null)
+	{
+		$ret = [];
+		foreach ($keys as $key) {
+			$ret[$key] = \Arr::get($array, $key, $default);
```
Comment (frankdejonge) [inline classes/arr.php]: this should be static:: instead of \Arr

File: classes/arr.php [hunk 2]
```diff
@@ -1207,4 +1207,24 @@ public static function next_by_value($array, $value, $get_value = true, $strict
...
+	 * @param   array    $array    the array containing the values
+	 * @param   array    $keys     list of keys (or indices) to return
+	 * @param   mixed    $default  value of missing keys; default null
+	 *
+	 * @return  array  An array containing the same set of keys provided.
+	 */
+	public function subset($array, $keys, $default=null)
+	{
+		$ret = [];
+		foreach ($keys as $key) {
```
Comment (frankdejonge) [inline classes/arr.php]: The bracket should be on a new line and a empty line before the foreach.

File: classes/arr.php [hunk 3]
```diff
@@ -1207,4 +1207,24 @@ public static function next_by_value($array, $value, $get_value = true, $strict
...
+			$ret[$key] = \Arr::get($array, $key, $default);
+		}
+		return $ret;
```
Comment (frankdejonge) [inline classes/arr.php:1231]: There should be a new line before this return statement.

**Comment by frankdejonge** [comment]
Looks good, can you update your code with the feedback I gave and request a pull for docs?

File: classes/arr.php [hunk 4]
```diff
@@ -1207,4 +1207,24 @@ public static function next_by_value($array, $value, $get_value = true, $strict
...
+	 *
+	 * Returns $default for missing keys, as with Arr::get()
+	 *
+	 * @param   array    $array    the array containing the values
+	 * @param   array    $keys     list of keys (or indices) to return
+	 * @param   mixed    $default  value of missing keys; default null
+	 *
+	 * @return  array  An array containing the same set of keys provided.
+	 */
+	public function subset($array, $keys, $default=null)
```
Comment (frankdejonge) [inline classes/arr.php]: $array and $keys should both be typehinted as array and the = should have spaces around it.

**Comment by Altreus** [comment]
Docs PR incoming...

**Comment by Altreus** [comment]
https://github.com/fuel/docs/pull/585

**Pull request merged**